clients/js/src/index.ts

```diff
@@ -0,0 +1,174 @@
+import { DefaultApi } from "./generated/api";
+import { Configuration } from "./generated/configuration";
+
+// a function to convert a non-Array object to an Array
+function toArray<T>(obj: T | Array<T>): Array<T> {
+  if (Array.isArray(obj)) {
+    return obj;
+  } else {
+    return [obj];
+  }
+}
+
+// a function to convert an array to array of arrays
+function toArrayOfArrays<T>(obj: Array<Array<T>> | Array<T>): Array<Array<T>> {
+  if (Array.isArray(obj[0])) {
+    return obj as Array<Array<T>>;
+  } else {
+    return [obj] as Array<Array<T>>;
+  }
+}
+
+export class Collection {
+  private name: string;
+  private api: DefaultApi;
+
+  constructor(name: string, api: DefaultApi) {
+    this.name = name;
+    this.api = api;
+  }
+
+  public async add(
+    ids: string | Array<any>,
+    embeddings: Array<any>,
+    metadatas?: Array<any> | object,
+    documents?: string | Array<any>,
+    increment_index: boolean = true,
+  ) {
+
+    const idsArray = toArray(ids);
+    const embeddingsArray = toArrayOfArrays(embeddings);
+
+    let metadatasArray;
+    if (metadatas === undefined) {
+      metadatasArray = undefined
+    } else {
+      metadatasArray = toArray(metadatas);
+    }
+
+    let documentsArray;
+    if (metadatas === undefined) {
+      documentsArray = undefined
+    } else {
+      documentsArray = toArray(metadatas);
```

Comment:
Looks like lines 50 and 53 are using `metadatas` instead of `documentsArray`

Reply:
@zzeleznick this is fixed on my local machine! :) thanks for reporting it. tests (as per usual) are very helpful.